trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -381,7 +348,102 @@ def spin(self,conditions):
         Va_ind_avg = va
         Vt_avg     = Wt
         Va_avg     = Wa
-          
+        
+        #------------------------------------------------------------------------------------------
+        # Now, if nonuniform freestream, analyze with the nonuniform velocities at propeller blade sections:
+        #------------------------------------------------------------------------------------------
+        
+        if nonuniform_freestream:             
```

Comment:
Rather than compute the entire case assuming there is a uniform freestream and then now checking. Put the if statement before.